plugins/modules/mso_schema_template_service_graph.py

```diff
@@ -0,0 +1,264 @@
+    host: mso_host
+    username: admin
+    password: SomeSecretPassword
+    schema: Schema1
+    template: Template1
+    service_graph: graph1
+    state: absent
+  delegate_to: localhost
+
+- name: Query a specific service graph
+  cisco.mso.mso_schema_template_service_graph:
+    host: mso_host
+    username: admin
+    password: SomeSecretPassword
+    schema: Schema1
+    template: Template1
+    service_graph: graph1
+    state: query
+  delegate_to: localhost
+
+- name: Query all service graphs
+  cisco.mso.mso_schema_template_service_graph:
+    host: mso_host
+    username: admin
+    password: SomeSecretPassword
+    schema: Schema1
+    template: Template1
+    state: query
+  delegate_to: localhost
+'''
+
+RETURN = r'''
+'''
+
+from ansible.module_utils.basic import AnsibleModule
+from ansible_collections.cisco.mso.plugins.module_utils.mso import MSOModule, mso_argument_spec, mso_node_spec
+
+
+def main():
+    argument_spec = mso_argument_spec()
+    argument_spec.update(
+        schema=dict(type='str', required=True),
+        template=dict(type='str', required=True),
+        service_graph_name=dict(type='str'),
+        service_graph_description=dict(type='str', default=''),
+        service_graph_display_name=dict(type='str'),
+        service_node_list=dict(type='list',dict='elements', options=mso_node_spec()),
+        node_filter=dict(type='str', choices=['allow-all', 'filters-from-contract']),
+        state=dict(type='str', default='present', choices=['absent', 'present', 'query']),
+    )
+
+    module = AnsibleModule(
+        argument_spec=argument_spec,
+        supports_check_mode=True,
+        required_if=[
+            ['state', 'absent', ['service_graph_name']],
+            ['state', 'present', ['service_graph_name', 'service_node_list']],
+        ],
+    )
+
+    schema = module.params.get('schema')
+    template = module.params.get('template').replace(' ', '')
+    service_graph_name = module.params.get('service_graph_name')
+    service_graph_display_name = module.params.get('service_graph_display_name')
+    service_graph_description = module.params.get('service_graph_description')
+    service_node_list = module.params.get('service_node_list')
+    node_filter = module.params.get('node_filter')
+    state = module.params.get('state')
+
+    mso = MSOModule(module)
+
+    # Get schema
+    schema_id, schema_path, schema_obj = mso.query_schema(schema)
+
+    # Get template
+    templates = [t.get('name') for t in schema_obj.get('templates')]
+    if template not in templates:
+        mso.fail_json(msg="Provided template '{template}' does not exist. Existing templates: {templates}".format(template=template,
+                                                                                                                  templates=', '.join(templates)))
+    template_idx = templates.index(template)
+
+    mso.existing = {}
+    service_graph_idx = None
+    index = 0
```

Comment:
what is it an index of? make the name clearer.

Reply:
ok